google/cloud/bigquery/table.py

```diff
@@ -126,45 +127,93 @@ def _view_use_legacy_sql_getter(table):
         return True
 
 
-class TableReference(object):
-    """TableReferences are pointers to tables.
+class TableBase:
```

Comment:
Let's either make this private (`_TableBase`) or document it + update relevant types in the `client` module.

Reply:
I don't see the end users using `TableBase` directly, it just serves as a central place where we keep the common functionality. An implementation detail, in other words.

Let's mark it as private.

Reply:
The contrarian point is that parameter types might use a public `TableBase` to signal the minimal contract that their arguments require, which is more focused than `Union[Table, TableRef, ...]`.  I don't see any case where `TableBase` as a return type make sense.

Reply:
After some thought I'm leaning towards keeping the base class private if the main reason for making it public is cleaner type annotations. I view type annotations more as an optional helper, and not as an end goal that would require adjusting the code just for the annotations' sake. As a bonus we still have an option to define our own `TableType` alias if we don't like `Union[Table, ...]`.

Or does anybody have a strong contrary opinion here?